docs/manual/other-topics/aws-lambda.md

```diff
@@ -0,0 +1,730 @@
+not used. This guide seeks to clarify some of these concepts so users of the library can properly
+configure `sequelize` for AWS Lambda and troubleshoot issues.
+
+## TL;DR
+
+If you just want to learn how to properly configure `sequelize`
+[connection pooling](./connection-pool.html) for AWS Lambda, all you need to know is that
+`sequelize` connection pooling does not get along well with AWS Lambda's Node.js runtime and it ends
+up causing more problems than it solves. Therefore, the most appropriate configuration is to **use
+pooling within the same invocation** and **avoid pooling across invocations** (i.e. close all
+connections at the end):
+
+```js
+const { Sequelize } = require("sequelize");
+
+let sequelize = null;
+
+async function loadSequelize() {
+  const sequelize = new Sequelize(/* (...) */, {
+    // (...)
+    pool: {
+      /*
+       * Lambda functions process one request at a time but your code may issue multiple queries
+       * concurrently. Be wary that `sequelize` has methods that issue 2 queries concurrently
+       * (e.g. `Model.findAndCountAll()`). Using a value higher than 1 allows concurrent queries to
+       * be executed in parallel rather than serialized. Careful with executing too many queries in
+       * parallel per Lambda function execution since that can bring down your database with an
+       * excessive number of connections.
+       *
+       * Ideally you want to choose a `max` number where this holds true:
+       * max * EXPECTED_MAX_CONCURRENT_LAMBDA_INVOCATIONS < MAX_ALLOWED_DATABASE_CONNECTIONS * 0.8
+       */
+      max: 2,
+      /*
+       * Set this value to 0 so connection pool eviction logic eventually cleans up all connections
+       * in the event of a Lambda function timeout.
+       */
+      min: 0,
+      /*
+       * Set this value to 0 so connections are eligible for cleanup immediately after they're
+       * returned to the pool.
+       */
+      idle: 0,
+      // Choose a small enough value that fails fast if a connection takes too long to be established.
+      acquire: 3000,
+      /*
+       * Ensures the connection pool attempts to be cleaned up automatically on the next Lambda
+       * function invocation, if the previous invocation timed out.
+       */
+      evict: CURRENT_LAMBDA_FUNCTION_TIMEOUT
+    }
+  });
+
+  // or `sequelize.sync()`
+  await sequelize.authenticate();
+
+  return sequelize;
+}
+
+module.exports.handler = async function (event, callback) {
+  // re-use the sequelize instance across invocations to improve performance
+  if (!sequelize) {
+    sequelize = await loadSequelize();
+  } else {
+    // restart connection pool to ensure connections are not re-used across invocations
+    sequelize.connectionManager.initPools();
+
+    // restore `getConnection()` if it has been overwritten by `close()`
+    if (sequelize.connectionManager.hasOwnProperty("getConnection")) {
+      delete sequelize.connectionManager.getConnection;
```

Comment:
Thanks for the awesome and detailed writeup @marcogrcr! 

I was testing out the `TL;DR` example code and the Typescript compiler was giving an `The operand of a 'delete' operator must be optional.` error in response to the following line:
```
delete sequelize.connectionManager.getConnection;
```

I know this is out of the scope of this PR, but maybe a better solution to the problem is to modify the [`getConnection()` function](https://github.com/sequelize/sequelize/blob/9634338d8f9a07ff151f70a91793b37547e2c23e/lib/dialects/abstract/connection-manager.js#L243) to check if `!this.pool` and throw an error. That way, we wouldn't need to get the [`close()` function](https://github.com/sequelize/sequelize/blob/9634338d8f9a07ff151f70a91793b37547e2c23e/lib/dialects/abstract/connection-manager.js#L110) to overwrite `getConnection()`. 

I would gladly make a PR if you deem fit @papb

Reply:
Hey @prestonlimlianjie:

Thanks for reviewing and the kind words!

I like your solution better. It's simple and it doesn't require changes to the TypeScript definitions. Happy to update this pull request if #13197 gets merged before this one.

BTW, I have returned you the favor reviewed #13197 with a couple of minor comments.

Reply:
Thanks for the PR review @marcogrcr - btw it seems like you're not on the Sequelize Slack org, you might want to join it especially since you're very passionate about Sequelize/they're looking for maintainers on the project! https://github.com/sequelize/sequelize#note-looking-for-maintainers